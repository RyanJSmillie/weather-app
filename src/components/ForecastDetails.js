import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecasts }) {
  const { date, temperature, humidity, wind, description } = forecasts;
  const formattedDate = new Date(date).toDateString();
  let windDirection = "";
  const { direction } = wind;

  if (direction === "n") {
    windDirection = "North";
  } else if (direction === "nne") {
    windDirection = "North North-East";
  } else if (direction === "ne") {
    windDirection = "North East";
  } else if (direction === "ene") {
    windDirection = "East North-East";
  } else if (direction === "e") {
    windDirection = "East";
  } else if (direction === "ese") {
    windDirection = "East South-East";
  } else if (direction === "se") {
    windDirection = "South East";
  } else if (direction === "sse") {
    windDirection = "South South-East";
  } else if (direction === "s") {
    windDirection = "South";
  } else if (direction === "ssw") {
    windDirection = "South South-West";
  } else if (direction === "sw") {
    windDirection = "South West";
  } else if (direction === "wsw") {
    windDirection = "West South-West";
  } else if (direction === "w") {
    windDirection = "West";
  } else if (direction === "wnw") {
    windDirection = "West North-West";
  } else if (direction === "nw") {
    windDirection = "North West";
  } else {
    windDirection = "North North-West";
  }

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__description">
        Conditions: {description}
      </div>
      <div className="forecast-details__maxTemp">
        Max Temperature: {temperature.max} &deg;C
      </div>
      <div className="forecast-details__minTemp">
        Min Temperature: {temperature.min} &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}</div>
      <div className="forecast-details__windSpeed">
        Wind Speed: {wind.speed} MPH
      </div>
      <div className="forecast-details__windDirection">
        Wind Direction: {windDirection}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecasts: PropTypes.shape({
    date: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    humidity: PropTypes.number.isRequired,

    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,

    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
