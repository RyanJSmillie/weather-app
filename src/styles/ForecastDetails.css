.forecast-details {
    margin: 30px;
    text-align: center;
    line-height: 25px;
}