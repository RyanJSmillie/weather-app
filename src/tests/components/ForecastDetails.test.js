import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      max: 22,
      min: 12,
    },
    description: "Clear",
    humidity: 15,
    wind: {
      speed: 15,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastDetails forecasts={validProps} />
    );

    expect(getAllByTestId("forecast-details")).toHaveLength(1);
  });
});
