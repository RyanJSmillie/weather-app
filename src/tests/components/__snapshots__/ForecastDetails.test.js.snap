// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__description"
    >
      Conditions: Clear
    </div>
    <div
      class="forecast-details__maxTemp"
    >
      Max Temperature: 22 °C
    </div>
    <div
      class="forecast-details__minTemp"
    >
      Min Temperature: 12 °C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 15
    </div>
    <div
      class="forecast-details__windSpeed"
    >
      Wind Speed: 15 MPH
    </div>
    <div
      class="forecast-details__windDirection"
    >
      Wind Direction: North
    </div>
  </div>
</DocumentFragment>
`;
